XRTK.SpatialPersistence/Packages/com.xrtk.spatial-persistence/Runtime/Interfaces/IMixedRealitySpatialPersistenceDataProvider.cs

```diff
@@ -0,0 +1,188 @@
+// Copyright (c) XRTK. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using UnityEngine;
+using XRTK.Definitions.SpatialPersistence;
+using XRTK.Definitions.Utilities;
+
+namespace XRTK.Interfaces.SpatialPersistence
+{
+    /// <summary>
+    /// Interface contract for specific identity provider implementations for use in the <see cref="IMixedRealitySpatialPersistenceSystem"/>.
+    /// </summary>
+    public interface IMixedRealitySpatialPersistenceDataProvider : IMixedRealityExtensionDataProvider
+    {
+        /// <summary>
+        /// The Instance Type for the Spatial Persistence Data Provider
+        /// </summary>
+        SystemType SpatialPersistenceType { get; }
```

Comment:
could we use an interface to define the spatial persistence type and then constrain the type here?